mirgecom/initializers.py

```diff
@@ -54,60 +54,62 @@
 from pytools.obj_array import make_obj_array
 from mirgecom.eos import IdealSingleGas
 from mirgecom.fluid import make_conserved
+from mirgecom.gas_model import make_fluid_state
 
 
-def initialize_flow_solution(actx, dcoll, gas_model, dd_bdry=None, pressure=None,
-                             temperature=None, density=None, velocity=None,
-                             mass_fractions=None, make_fluid_state=False):
+def initialize_flow_solution(actx, nodes, gas_model=None, eos=None,
+        pressure=None, temperature=None, density=None, velocity=None,
+        species_mass_fractions=None, return_fluid_state=False):
     """Create a fluid CV/state from a set of minimal input data."""
-    if gas_model is None:
-        raise ValueError("Gas model is required to create a FluidState.")
+    if gas_model is None and eos is None:
+        raise ValueError("Must provide 1 of (gas_model, eos).")
+    if eos is None:
+        eos = gas_model.eos
+    if return_fluid_state and gas_model is None:
+        raise ValueError("Must provide gas_model to create a fluid state.")
```

Comment:
What if gas_model and eos are both None?  EOS should always come from gas_model if it exists, can we do something like:
```suggestion
    if gas_model is not None:
       	if eos is not None:
       	    warn("Specifying both EOS and gas_model is overspecified")
       	 eos = gas_model.eos
    elif return_fluid_state:
        raise ValueError("Must provide gas_model to create a fluid state.")
```

Reply:
Following comment below, this is history.